lib/p5.play.js

```diff
@@ -4338,4 +3909,1314 @@ p5.prototype._warn = function(message) {
+   * Modifies this transform and returns it.
+   * @method mult
+   * @param {p5.Transform2D|Float32Array|Array.<number>} other
+   * @return {p5.Transform2D}
+   */
+  p5.Transform2D.prototype.mult = function(other) {
+    return p5.Transform2D.mult(this, other, this);
+  };
+
+  /**
+   * Modify this transform, translating it by a certain amount.
+   * Returns this transform.
+   * @method translate
+   * @return {p5.Transform2D}
+   * @example
+   *     // Two different ways to call this method.
+   *     var t = new p5.Transform();
+   *     // 1. Two numbers
+   *     t.translate(x, y);
+   *     // 2. One vector
+   *     t.translate(new p5.Vector(x, y));
+   */
+  p5.Transform2D.prototype.translate = function(arg0, arg1) {
+    var x, y;
+    if (arg0 instanceof p5.Vector) {
+      x = arg0.x;
+      y = arg0.y;
+    } else if (typeof arg0 === 'number' && typeof arg1 === 'number') {
+      x = arg0;
+      y = arg1;
+    } else {
+      var args = '';
+      for (var i = 0; i < arguments.length; i++) {
+        args += arguments[i] + ', ';
+      }
+      throw new TypeError('Invalid arguments to Transform2D.translate: ' + args);
+    }
+    return p5.Transform2D.mult([
+      1, 0, x,
+      0, 1, y
+    ], this, this);
+  };
+
+  /**
+   * Retrieve the resolved translation of this transform.
+   * @method getTranslation
+   * @return {p5.Vector}
+   */
+  p5.Transform2D.prototype.getTranslation = function() {
+    return new p5.Vector(this[2], this[5]);
+  };
+
+  /**
+   * Modify this transform, scaling it by a certain amount.
+   * Returns this transform.
+   * @method scale
+   * @return {p5.Transform2D}
+   * @example
+   *     // Three different ways to call this method.
+   *     var t = new p5.Transform();
+   *     // 1. One scalar value
+   *     t.scale(uniformScale);
+   *     // 1. Two scalar values
+   *     t.scale(scaleX, scaleY);
+   *     // 2. One vector
+   *     t.translate(new p5.Vector(scaleX, scaleY));
+   */
+  p5.Transform2D.prototype.scale = function(arg0, arg1) {
+    var sx, sy;
+    if (arg0 instanceof p5.Vector) {
+      sx = arg0.x;
+      sy = arg0.y;
+    } else if (typeof arg0 === 'number' && typeof arg1 === 'number') {
+      sx = arg0;
+      sy = arg1;
+    } else if (typeof arg0 === 'number') {
+      sx = arg0;
+      sy = arg0;
+    } else {
+      throw new TypeError('Invalid arguments to Transform2D.scale: ' + arguments);
+    }
+    return p5.Transform2D.mult([
+      sx, 0, 0,
+      0, sy, 0
+    ], this, this);
+  };
+
+  /**
+   * Retrieve the scale vector of this transform.
+   * @method getScale
+   * @return {p5.Vector}
+   */
+  p5.Transform2D.prototype.getScale = function() {
+    var a = this[0], b = this[1],
+        c = this[3], d = this[4];
+    return new p5.Vector(
+      sign(a) * Math.sqrt(a*a + b*b),
+      sign(d) * Math.sqrt(c*c + d*d)
+    );
+  };
+
+  /*
+   * Return -1, 0, or 1 depending on whether a number is negative, zero, or positive.
+   */
+  function sign(x) {
```

Comment:
Polyfilled - [`Math.sign` is a thing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/sign) but only in Chrome+Firefox for now.